src/docker.c

```diff
@@ -956,43 +1202,94 @@ static int get_network_stats(yajl_val node, container_stats_t *stats) {
   return successes;
 }
 
-// Parses the JSON response from the Docker STATS API and extracts metrics
-// for memory, disk, CPU and networking interfaces.
-static void extract_stats_from_response(char *response_buffer,
+/* Purpose: Extract all the stats/metrics from the JSON response.
+ *
+ * Params: (char *) response_buffer: string holding the Docker stats response.
+ *          (constainer_stats_t **) stats: Pointer to the container_stats_t
+ *                                          struct which holds all the metrics
+ *                                          for a given container once the
+ *                                          function returns.
+ *          (const char *) container_id: 64-character container ID.
+ *
+ * Returns: int
+ *          0 in case of success.
+ *          -1 in case of partial success.
+ *          -2 in case of error / complete failure.
+ **/
+static int extract_stats_from_response(char *response_buffer,
     container_stats_t **stats, const char *container_id) {
   yajl_val node;
   char errbuf[1024];
+  int errors = 0;
   container_stats_t *ptr = (container_stats_t *)
       calloc(1, sizeof(container_stats_t));
   if (ptr == NULL) {
     ERROR("docker: extract_stats_from_response: malloc failed!");
-    return;
+    return -2;
```

Comment:
Could we define named constants for the return codes instead of hard coding numbers?